web/source/dom/domEventHandlers.ts

```diff
@@ -641,69 +641,10 @@ namespace com.keyman.dom {
         target.scrollInput();
         // nextSibling - the scrollbar element.
       } else if(tTarg != scroller.nextSibling) { // Otherwise, if clicked on text in SPAN, set at touch position
-        var caret,cp,cpMin,cpMax,x,y,dy,yRow,iLoop;
-        caret=scroller.childNodes[1]; //caret span
-        cpMin=0;
-        cpMax=target.getText()._kmwLength();
-        cp=target.getTextCaret();
-        dy=document.body.scrollTop;
-
-        // Vertical scrolling
-        if(target.base instanceof target.base.ownerDocument.defaultView.HTMLTextAreaElement) {
-          yRow=Math.round(target.base.offsetHeight/(target.base as HTMLTextAreaElement).rows);
-          for(iLoop=0; iLoop<16; iLoop++)
-          {
-            y=dom.Utils.getAbsoluteY(caret)-dy;  //top of caret
-            if(y > touchY && cp > cpMin && cp != cpMax) {cpMax=cp; cp=Math.round((cp+cpMin)/2);}
-            else if(y < touchY-yRow && cp < cpMax && cp != cpMin) {cpMin=cp; cp=Math.round((cp+cpMax)/2);}
-            else break;
-            target.setTextCaret(cp);
-          }
-
-          while(dom.Utils.getAbsoluteY(caret)-dy > touchY && cp > cpMin) {
-            target.setTextCaret(--cp);
-          }
-
-          while(dom.Utils.getAbsoluteY(caret)-dy < touchY-yRow && cp < cpMax) {
-            target.setTextCaret(++cp);
-          }
```

Comment:
For comparison with the other block, re: `getAbsoluteY` motivating the `clientY` -> `pageY` change.

This block still exists; it's just been refactored into the new `performCaretSearch` function.  Along with some light refactoring to reformat and clarify the code, as it was somewhat dense in its original form.